### ISSUE TITLE
Tests failing with Pydantic 2.7.0 update

### ISSUE THREAD

**Issue by mferrera** [equinor/fmu-dataio#600]
A new version was released yesterday.

https://github.com/equinor/fmu-dataio/actions/runs/8654958823/job/23733116877

They all seem to relate to enum values

```
yaml.representer.RepresenterError: ('cannot represent an object', <ContentEnum.depth: 'depth'>)
```

This may relate to the following Pydantic change:

```
Add enum and type to the JSON schema for single item literals
```

which also changes the schema a bit.

**Comment by mferrera** [comment]
Test case that represents the issue. I think there are two parts at play here.

```python
from enum import Enum
from typing import Literal, Optional, Union

from pydantic import BaseModel, Field, RootModel
from typing_extensions import Annotated


class SomeEnum(str, Enum):
    FOO = "foo"
    BAR = "bar"
    BAZ = "baz"


class SomeModel(BaseModel):
    some_enum: SomeEnum = Field(description="The contents")
    another_enum: Optional[SomeEnum] = Field(
        default=SomeEnum.FOO, validate_default=True
    )


class FooModel(SomeModel):
    some_enum: Literal[SomeEnum.FOO]


class BarModel(SomeModel):
    some_enum: Literal[SomeEnum.BAR]


class AnyModel(RootModel):
    root: Annotated[Union[FooModel, BarModel], Field(discriminator="some_enum")]


foo_model = FooModel(some_enum=SomeEnum.FOO)
print("foo_model:", foo_model.model_dump(mode="json"))
# {'some_enum': 'foo', 'another_enum': 'foo'}

any_model = AnyModel.model_validate(foo_model)
print("any_model:", any_model.model_dump(mode="json"))
# {'some_enum': <SomeEnum.FOO: 'foo'>, 'another_enum': 'foo'}
```

Setting `model_config = ConfigDict(use_enum_values=True)` does not help in this case. However, changing the enum to not use the str mixin does:

```python
class SomeEnum(Enum):
    FOO = "foo"
    BAR = "bar"
    ...

...

foo_model = FooModel(some_enum=SomeEnum.FOO)
print("foo_model:", foo_model.model_dump(mode="json"))
# {'some_enum': 'foo', 'another_enum': 'foo'}

any_model = AnyModel.model_validate(foo_model)
print("any_model:", any_model.model_dump(mode="json"))
# {'some_enum': 'foo', 'another_enum': 'foo'}
```

It seems more probable that this is a Pydantic bug. Not using `str` in the enum opens other errors so it cannot be fixed just with that.

**Comment by mferrera** [comment]
Most likely related to https://github.com/pydantic/pydantic/issues/9235

**Comment by mferrera** [comment]
Will be resolved by #611
